Mathlib/Order/Bounds/Basic.lean

```diff
@@ -125,6 +125,37 @@ theorem isLUB_congr (h : upperBounds s = upperBounds t) : IsLUB s a ↔ IsLUB t
 theorem isGLB_congr (h : lowerBounds s = lowerBounds t) : IsGLB s a ↔ IsGLB t a := by
   rw [IsGLB, IsGLB, h]
 
+@[simp] lemma Dominated.of_subset (hst : s ⊆ t) : Dominated s t := fun a ha ↦ ⟨a, hst ha, le_rfl⟩
+
+@[simp] lemma Codominated.of_subset (hst : s ⊆ t) : Codominated s t :=
+  fun a ha ↦ ⟨a, hst ha, le_rfl⟩
+
+alias HasSubset.Subset.dominated := Dominated.of_subset
+alias HasSubset.Subset.codominated := Codominated.of_subset
+
+@[simp, refl] protected lemma Dominated.rfl : Dominated s s := .of_subset .rfl
+@[simp, refl] protected lemma Codominated.rfl : Codominated s s := .of_subset .rfl
+
+protected lemma Dominated.trans (hst : Dominated s t) (htu : Dominated t u) :
+    Dominated s u :=
+  fun _a ha ↦ let ⟨_b, hb, hab⟩ := hst ha; let ⟨c, hc, hbc⟩ := htu hb; ⟨c, hc, hab.trans hbc⟩
+
+protected lemma Codominated.trans (hst : Codominated s t) (htu : Codominated t u) :
+    Codominated s u :=
+  fun _a ha ↦ let ⟨_b, hb, hba⟩ := hst ha; let ⟨c, hc, hcb⟩ := htu hb; ⟨c, hc, hcb.trans hba⟩
+
+protected lemma Dominated.mono_left (hst : s ⊆ t) (htu : Dominated t u) : Dominated s u :=
```

Comment:
Should this be antitone rather than `mono`?

Reply:
It's about half `mono` half `anti` in mathlib, we can't seem to make up our mind. I think it doesn't really matter either way so long as it doesn't create ambiguity.